packages/core/src/routes/sign-in-experience.guard.test.ts

```diff
@@ -0,0 +1,155 @@
+
+import signInExperiencesRoutes from './sign-in-experience';
+
+jest.mock('@/queries/sign-in-experience', () => ({
+  updateDefaultSignInExperience: jest.fn(
+    async (data: Partial<CreateSignInExperience>): Promise<SignInExperience> => ({
+      ...mockSignInExperience,
+      ...data,
+    })
+  ),
+}));
+
+const signInExperienceRequester = createRequester({ authedRoutes: signInExperiencesRoutes });
+
+const expectPatchResponseStatus = async (signInExperience: any, status: number) => {
+  const response = await signInExperienceRequester.patch('/sign-in-exp').send(signInExperience);
+  expect(response.status).toEqual(status);
+};
+
+describe('branding', () => {
+  const invalidColors = [
+    undefined,
+    '',
+    '#',
+    '#1',
+    '#2B',
+    '#3cZ',
+    '#4D9e',
+    '#5f80E',
+    '#6GHiXY',
+    '#78Cb5dA',
+    'rgb(0,13,255)',
+  ];
+
+  const validColors = ['#aB3', '#169deF'];
+
+  describe('primaryColor', () => {
+    test.each(validColors)('%p should succeed', async (primaryColor) => {
+      const signInExperience = { branding: { ...mockBranding, primaryColor } };
+      await expectPatchResponseStatus(signInExperience, 200);
+    });
+    test.each(invalidColors)('%p should fail', async (primaryColor) => {
+      const signInExperience = { branding: { ...mockBranding, primaryColor } };
+      await expectPatchResponseStatus(signInExperience, 400);
+    });
+  });
+
+  describe('backgroundColor', () => {
+    test.each(validColors)('%p should succeed', async (backgroundColor) => {
+      const signInExperience = { branding: { ...mockBranding, backgroundColor } };
+      await expectPatchResponseStatus(signInExperience, 200);
+    });
+    test.each(invalidColors)('%p should fail', async (backgroundColor) => {
+      const signInExperience = { branding: { ...mockBranding, backgroundColor } };
+      await expectPatchResponseStatus(signInExperience, 400);
+    });
+  });
+
+  describe('darkPrimaryColor', () => {
+    test.each(validColors)('%p should succeed', async (darkPrimaryColor) => {
+      const signInExperience = { branding: { ...mockBranding, darkPrimaryColor } };
+      await expectPatchResponseStatus(signInExperience, 200);
+    });
+    test.each(invalidColors)('%p should fail', async (darkPrimaryColor) => {
+      const signInExperience = { branding: { ...mockBranding, darkPrimaryColor } };
+      await expectPatchResponseStatus(signInExperience, 400);
+    });
+  });
+
+  describe('darkBackgroundColor', () => {
+    test.each(validColors)('%p should succeed', async (darkBackgroundColor) => {
+      const signInExperience = { branding: { ...mockBranding, darkBackgroundColor } };
+      await expectPatchResponseStatus(signInExperience, 200);
+    });
+    test.each(invalidColors)('%p should fail', async (darkBackgroundColor) => {
+      const signInExperience = { branding: { ...mockBranding, darkBackgroundColor } };
+      await expectPatchResponseStatus(signInExperience, 400);
+    });
+  });
```

Comment:
could merge all color keys into a single loop? since all cases are basically testing the same logic

Reply:
Updated. like this?

```ts
  describe('colors', () => {
    test.each(validColors)('%p should succeed', async (validColor) => {
      await expectPatchResponseStatus(
        { branding: { ...mockBranding, primaryColor: validColor } },
        200
      );
      await expectPatchResponseStatus(
        { branding: { ...mockBranding, backgroundColor: validColor } },
        200
      );
      await expectPatchResponseStatus(
        { branding: { ...mockBranding, darkPrimaryColor: validColor } },
        200
      );
      await expectPatchResponseStatus(
        { branding: { ...mockBranding, darkBackgroundColor: validColor } },
        200
      );
    });
    test.each(invalidColors)('%p should fail', async (invalidColor) => {
      await expectPatchResponseStatus(
        { branding: { ...mockBranding, primaryColor: invalidColor } },
        400
      );
      await expectPatchResponseStatus(
        { branding: { ...mockBranding, backgroundColor: invalidColor } },
        400
      );
      await expectPatchResponseStatus(
        { branding: { ...mockBranding, darkPrimaryColor: invalidColor } },
        400
      );
      await expectPatchResponseStatus(
        { branding: { ...mockBranding, darkBackgroundColor: invalidColor } },
        400
      );
    });
  });
```